getqr.sh

```diff
@@ -1,18 +1,15 @@
-#!/bin/bash
-set -o xtrace
+#!/bin/sh -x
 signal-desktop --no-sandbox & # run in the background
 pid=$! # $! is the pid of the last run process
-sleep 1
-until [ "${window_id-}" ]; do # that substitution will be empty instead of an error if window_id is unset
+sleep 2
+
+# wmctrl -lp is a table of window_id, desktop number, pid, user, window title
+while window_id=$(wmctrl -lp | awk -v proc="$pid" '{if ($3 == proc) {print $1; exit 1}'); do
```

Comment:
How does this work?

Reply:
How'd you mean?
`window_id=$(wmctrl -lp | awk -v proc="$pid" '{if ($3 == proc) {print $1; exit 1}')` is has the exit code of the subshell, and the subshell has the exit code of the final pipeline member and awk exits true unless it finds something

Reply:
When I run this, I get:


+ xwd -id
+ tee qr.xwd
+ zbarimg -q xwd:-
+ tee zbar_output
+ cut -d: -f2-
+ tee signal_uri
xwd: error: Invalid window id format: .
ERROR: unable to read image header `-' @ error/xwd.c/ReadXWDImage/230

Reply:
uhoh, thats because i did a poopy and missed a final closing `}`. fixed.

Reply:
perfect